File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/DocumentationUnitCreationParameters.java
package de.bund.digitalservice.ris.caselaw.domain;

import de.bund.digitalservice.ris.caselaw.domain.court.Court;
import de.bund.digitalservice.ris.caselaw.domain.lookuptable.documenttype.DocumentType;
import jakarta.validation.constraints.PastOrPresent;
import java.time.LocalDate;
import lombok.Builder;

@Builder(toBuilder = true)
public record DocumentationUnitCreationParameters(
    DocumentationOffice documentationOffice,
    Court court,
    DocumentType documentType,
    @PastOrPresent LocalDate decisionDate,
    String fileNumber,
    Reference reference) {}

File: frontend/src/types/documentationUnitCreationParameters.ts
import { Court } from "@/domain/court"
import DocumentationOffice from "@/domain/documentationOffice"
import { DocumentType } from "@/domain/documentType"
import Reference from "@/domain/reference"
import { Source } from "@/domain/source"

export type DocumentationUnitCreationParameters = {
  documentationOffice?: DocumentationOffice
  documentType?: DocumentType
  decisionDate?: string
  fileNumber?: string
  court?: Court
  reference?: Reference
  sources?: Source[]
}
